stingray/fourier.py

```diff
@@ -0,0 +1,1286 @@
+        sub-band periodogram
+    P2 : float `np.array`
+        reference-band periodogram
+    P1noise : float
+        Poisson noise level of the sub-band periodogram
+    P2noise : float
+        Poisson noise level of the reference-band periodogram
+    Nave : int
+        number of intervals that have been averaged to obtain the input spectra
+
+    """
+    new_coherence = np.ones_like(P1)
+    old_coherence = np.zeros_like(P1)
+    count = 0
+    while not np.allclose(new_coherence, old_coherence, atol=0.01) and count < 40:
+        # TODO: make it only iterate over the places at low coherence
+        old_coherence = new_coherence
+        bsq = bias_term(C, P1, P2, P1noise, P2noise, Nave, intrinsic_coherence=new_coherence)
+        # old_coherence = new_coherence
+        den = (P1 - P1noise) * (P2 - P2noise)
+        num = (C * C.conj()).real - bsq
+        num[num < 0] = (C * C.conj()).real[num < 0]
+        new_coherence = num / den
+        count += 1
+
+    return new_coherence
+
+
+def error_on_averaged_cross_spectrum(C, Ps, Pr, Nave, Psnoise, Prnoise, common_ref="False"):
+    """Error on cross spectral quantities, From Ingram 2019.
+
+    Parameters
+    ----------
+    C : complex `np.array`
+        cross spectrum
+    Ps : float `np.array`
+        sub-band periodogram
+    Pr : float `np.array`
+        reference-band periodogram
+    Psnoise : float
+        Poisson noise level of the sub-band periodogram
+    Prnoise : float
+        Poisson noise level of the reference-band periodogram
+    Nave : int
+        number of intervals that have been averaged to obtain the input spectra
+
+    Other Parameters
+    ----------------
+    common_ref : bool, default False
+        Are data in the sub-band also included in the reference band?
+
+    Returns
+    -------
+    dRe : float `np.array`
+        Error on the real part of the cross spectrum
+    dIm : float `np.array`
+        Error on the imaginary part of the cross spectrum
+    dphi : float `np.array`
+        Error on the angle (or phase lag)
+    dG : float `np.array`
+        Error on the modulus of the cross spectrum
+
+    """
+    twoN = 2 * Nave
+    if common_ref:
+        Gsq = (C * C.conj()).real
+        bsq = bias_term(C, Ps, Pr, Psnoise, Prnoise, Nave)
+        frac = (Gsq - bsq) / (Pr - Prnoise)
+        PoN = Pr / twoN
+
+        # Eq. 18
+        dRe = dIm = dG = np.sqrt(PoN * (Ps - frac))
+        # Eq. 19
+        dphi = np.sqrt(PoN * (Ps / (Gsq - bsq) - 1 / (Pr - Prnoise)))
+
+    else:
+        PrPs = Pr * Ps
+        dRe = np.sqrt((PrPs + C.real ** 2 - C.imag ** 2) / twoN)
+        dIm = np.sqrt((PrPs - C.real ** 2 + C.imag ** 2) / twoN)
+        gsq = raw_coherence(C, Ps, Pr, Psnoise, Prnoise, Nave)
+        dphi = np.sqrt((1 - gsq) / (2 * gsq ** 2 * Nave))
+        dG = np.sqrt(PrPs / Nave)
+
+    return dRe, dIm, dphi, dG
+
+
+def cross_to_covariance(C, Pr, Prnoise, delta_nu):
+    """Convert a cross spectrum into a covariance.
```

Comment:
So I feel like there are starting to be a *lot* of parameters that have capital letters. I'm not a big fan of that practice, but I'm happy to discuss a policy change more broadly if there are good arguments for it.

Reply:
Changed all to lowercase